backend/pkg/api/data_access/notifications.go

```diff
@@ -1845,11 +1841,21 @@ func (d *DataAccessService) GetNotificationSettingsDashboards(ctx context.Contex
 	settingsMap := make(map[string]*NotificationSettingsDashboardsInfo)
 
 	for _, event := range events {
-		eventSplit := strings.Split(event.Filter, ":")
-		if len(eventSplit) != 3 {
+		eventFilterSplit := strings.Split(event.Filter, ":")
+		if len(eventFilterSplit) != 3 {
 			continue
 		}
-		dashboardType := eventSplit[0]
+		dashboardType := eventFilterSplit[0]
+
+		eventNameSplit := strings.Split(string(event.Name), ":")
+		if len(eventNameSplit) != 2 && dashboardType == ValidatorDashboardEventPrefix {
+			return nil, nil, fmt.Errorf("invalid event name formatting for val dashboard notification: expected {network:event_name}, got %v", event.Name)
+		}
+
+		eventName := event.Name
+		if len(eventNameSplit) == 2 {
+			eventName = types.EventName(eventNameSplit[1])
+		}
```

Comment:
I implemented this under the assumption that the val dashboard events have the format {network:event_name} and acc dashboard events will have {event_name}.